planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/util/YAMLoptimizerParser.java

```diff
@@ -714,6 +698,21 @@ public   static String getInitialElementName(Map<String, Object> appMap) {
       return groupsMap;
    }
 
+   public static Map<String, Object> getTypesMapFromAppMap(Map<String, Object> appMap) {
+      Map<String, Object> typesMap;
+      try {
+         if (log.isDebugEnabled()) {
```

Comment:
I am not sure if it is useful `log.isDebugEnabled()` here. Take a look [here](http://stackoverflow.com/questions/963492/in-log4j-does-checking-isdebugenabled-before-logging-improve-performance)